.github/workflows/autoupdate.yml

```diff
@@ -0,0 +1,61 @@
+name: Autoupdate
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+
+jobs:
+  autoupdate:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: |
+        sudo pip3 install setuptools wheel
+        sudo pip3 install yq
+        pip3 install -r scripts/requirements.txt
+    - name: Run autoupdate
+      run: |
+        git config user.email "actions@github.com"
+        git config user.name "Github Actions"
+        git fetch origin
+        
+        for file in manifests/*
+        do
+          git reset --hard master
+          ./scripts/autoupdate.py "$file"
+          
+          if [[ `git status --porcelain` ]]; then
```

Comment:
turns out this was correct, we want to (supposedly) start a subshell and check if there's anything on stdout. The exit code is 0 either way.

Guess that's what i get from stealing commands from stackoverflow.

Reply:
You could pipe to grep and test that:
```bash
if git status --porcelain | grep -q "."; then
  # dirty dir
else
  # clean dir
```
Note that grep doesn't like it if you pass _nothing_, so you wouldn't want to do `if git status --porcelain | grep -v "."; then` to detect the case where the directory is clean. You'd negate the command (`if ! git...`), rather than the grep match for that scenario.

Reply:
That would work, but is it better than this?